ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -166,33 +167,33 @@ extraFiles ghcFlavor =
     , "ghc-lib/stage1/compiler/build/primop-vector-tys-exports.hs-incl"
     , "ghc-lib/stage1/compiler/build/primop-vector-tys.hs-incl"
     , "ghc-lib/stage1/compiler/build/primop-vector-uniques.hs-incl"
-    , "ghc-lib/stage0/compiler/build/Fingerprint.hs"
+    ] ++ ["ghc-lib/stage0/compiler/build/Fingerprint.hs" | ghcFlavor /= GhcMaster] ++
     -- Be careful not to add these files to a ghc-lib.cabal, just
     -- ghc-lib-parser.cabal.
-    , "ghc-lib/stage1/compiler/build/Config.hs"
+    [ "ghc-lib/stage1/compiler/build/Config.hs"
     , "ghc-lib/stage0/compiler/build/Parser.hs"
     , "ghc-lib/stage0/compiler/build/Lexer.hs"
     ] ++
     ["ghc-lib/stage0/libraries/ghc-boot/build/GHC/Version.hs" | ghcFlavor == GhcMaster ]
 
 -- | The C headers shipped with ghc-lib. These globs get glommed onto
 -- the 'extraFiles' above as 'extra-source-files'.
-cHeaders :: [String]
-cHeaders =
+cHeaders :: GhcFlavor -> [String]
+cHeaders ghcFlavor =
   [ "includes/ghcconfig.h"
   , "includes/MachDeps.h"
   , "includes/CodeGen.Platform.hs"
-  , "compiler/nativeGen/*.h"
-  , "compiler/utils/*.h"
-  , "compiler/*.h"
-  ]
+  , "compiler/HsVersions.h"
```

Comment:
Are `HsVersions` and `Unique` the only headers in `compiler` for GHC 8.8 or why are we switching from a wildcard for 8.8 to listing the two explicitly?

Reply:
I looked into it. Yes, they are the only headers in 8.8. I'm implementing a general policy of removing wild-cards (following on from what we learned with our rts related include work).